### ISSUE TITLE
Install errors if CONDA_BUILD_SYSROOT is not defined.

### ISSUE THREAD

**Issue by chebee7i** [conda-forge/nvcc-feedstock#40]
<!--
Thanks for reporting your issue.
Please fill out the sections below.
-->
Issue:

The `nvcc-linux_64` install assumes a variable `CONDA_BUILD_SYSROOT` exists, but it does not on my setup (conda 4.8.4 with/without conda-build installed).

```console
$ conda install -c pytorch nvcc_linux-64                                               
Collecting package metadata (current_repodata.json): done                              
Solving environment: done                                                              
                                                                                       
                                                                                       
## Package Plan ##                                                                     
                                                                                       
  environment location: /dev/shm/envs/hello                                          
                                                                                       
  added / updated specs:                                                               
    - nvcc_linux-64                                                                    
                                                                                       
                                                                                       
The following packages will be downloaded:                                             
                                                                                       
    package                    |            build                                      
    ---------------------------|-----------------                                      
    nvcc_linux-64-10.2         |       hc6a2c23_6           7 KB  conda-forge          
    ------------------------------------------------------------                       
                                           Total:           7 KB                       
                                                                                       
The following NEW packages will be INSTALLED:                                          
                                                                                       
  nvcc_linux-64      conda-forge/linux-64::nvcc_linux-64-10.2-hc6a2c23_6               
                                                                                       
                                                                                       
Proceed ([y]/n)? y                                                                     
Downloading and Extracting Packages
nvcc_linux-64-10.2   | 7 KB      | ############################################################################ | 100%
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
ln: failed to create symbolic link ‘/lib/libcuda.so’: Permission denied
```

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list
# packages in environment at /dev/shm/envs/hello:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main
_tflow_select             2.3.0                       mkl
absl-py                   0.9.0                    py38_0
agate                     1.6.1                    py38_2
agate-dbf                 0.2.1                      py_0
agate-excel               0.2.3                      py_0
agate-sql                 0.5.4                      py_0
alembic                   1.4.2                      py_0
astropy                   4.0.1.post1      py38h7b6447c_1
astunparse                1.6.3                      py_0
attrs                     19.3.0                     py_0
autobahn                  20.7.1                     py_0    conda-forge
automat                   0.8.0                      py_0
babel                     2.8.0                      py_0
backcall                  0.2.0                      py_0
bcrypt                    3.1.7            py38h7b6447c_1
beautifulsoup4            4.9.1                    py38_0
blas                      1.0                         mkl
bleach                    3.1.5                      py_0
blinker                   1.4                      py38_0
bokeh                     2.1.1                    py38_0
brotlipy                  0.7.0           py38h7b6447c_1000
bzip2                     1.0.8                h7b6447c_0
c-ares                    1.15.0            h7b6447c_1001
ca-certificates           2020.6.24                     0
cachetools                4.1.0                      py_1
cairo                     1.14.12              h8948797_3
catalogue                 1.0.0                    py38_1
certifi                   2020.6.20                py38_0
cffi                      1.14.0           py38h2e261b9_0
chardet                   3.0.4                 py38_1003
click                     7.1.2                      py_0
clickhouse-driver         0.1.4            py38h1e0a361_0    conda-forge
clickhouse-sqlalchemy     0.1.4                      py_0    conda-forge
cloudpickle               1.5.0                      py_0
conda                     4.8.4                    py38_0
conda-build               3.18.11                  py38_0
conda-package-handling    1.6.1            py38h7b6447c_0
configparser              5.0.0                      py_0
constantly                15.1.0                     py_0
coverage                  5.0              py38h7b6447c_0
croniter                  0.3.30                     py_0
cryptography              2.9.2            py38h1ba5d50_0
csvkit                    1.0.4                    py38_0
cvxopt                    1.2.0            py38hfa32c7d_0
cycler                    0.10.0                   py38_0
cymem                     2.0.3            py38he6710b0_0
cython-blis               0.4.1            py38h7b6447c_1
cytoolz                   0.10.1           py38h7b6447c_0
dask                      2.20.0                     py_0
dask-core                 2.20.0                     py_0
dbfread                   2.0.7                    py38_0
dbus                      1.13.16              hb2f20db_0
decorator                 4.4.2                      py_0
defusedxml                0.6.0                      py_0
dill                      0.3.2                      py_0
distributed               2.20.0                   py38_0
dpkt                      1.9.3              pyh9f0ad1d_0    conda-forge
drmaa                     0.7.9                    py38_0
entrypoints               0.3                      py38_0
et_xmlfile                1.0.1                   py_1001
expat                     2.2.9                he6710b0_2
fastcache                 1.1.0            py38h7b6447c_0
filelock                  3.0.12                     py_0
flake8                    3.8.3                      py_0
flaky                     3.6.1                      py_0
flask                     1.1.2                      py_0
fontconfig                2.13.0               h9420a91_0
freetype                  2.10.2               h5ab3b9f_0
fribidi                   1.0.9                h7b6447c_0
fsspec                    0.7.4                      py_0
future                    0.18.2                   py38_1
gast                      0.3.3                      py_0
gevent                    20.6.2           py38h7b6447c_0
glib                      2.63.1               h5a9c865_0
glob2                     0.7                        py_0
glpk                      4.65                 h3ceedfd_2
gmp                       6.1.2                h6c8ec71_1
gmpy2                     2.0.8            py38hd5f6e3b_3
google-auth               1.17.2                     py_0
google-auth-oauthlib      0.4.1                      py_2
google-pasta              0.2.0                      py_0
graphite2                 1.3.14               h23475e2_0
graphviz                  2.40.1               h21bd128_2
greenlet                  0.4.16           py38h7b6447c_0
grpcio                    1.27.2           py38hf8bcb03_0
gsl                       2.4                  h14c3975_4
gst-plugins-base          1.14.0               hbbd80ab_1
gstreamer                 1.14.0               hb453b48_1
h5py                      2.10.0           py38h7918eee_0
harfbuzz                  1.8.8                hffaf4a1_0
hdf5                      1.10.4               hb1b8bf9_0
heapdict                  1.0.1                      py_0
hyperlink                 19.0.0                     py_0
icu                       58.2                 he6710b0_3
idna                      2.10                       py_0
imageio                   2.9.0                      py_0
importlib-metadata        1.7.0                    py38_0
importlib_metadata        1.7.0                         0
incremental               17.5.0                   py38_0
influxdb                  5.3.0              pyh9f0ad1d_0    conda-forge
intel-openmp              2020.1                      217
ipdb                      0.13.0             pyh9f0ad1d_0    conda-forge
ipykernel                 5.3.3            py38h5ca1d4c_0
ipython                   7.16.1           py38h5ca1d4c_0
ipython_genutils          0.2.0                    py38_0
ipywidgets                7.5.1                      py_0
isodate                   0.6.0                      py_1
itsdangerous              1.1.0                      py_0
jdcal                     1.4.1                      py_0
jedi                      0.17.1                   py38_0
jinja2                    2.11.2                     py_0
joblib                    0.16.0                     py_0
jpeg                      9b                   h024ee3a_2
json5                     0.9.5                      py_0
jsonschema                3.0.2                    py38_0
jupyter                   1.0.0                    py38_7
jupyter_client            6.1.6                      py_0
jupyter_console           6.1.0                      py_0
jupyter_core              4.6.3                    py38_0
jupyterlab                2.1.5                      py_0
jupyterlab_server         1.2.0                      py_0
keras                     2.4.3                         0
keras-base                2.4.3                      py_0
keras-preprocessing       1.1.0                      py_1
kiwisolver                1.2.0            py38hfd86e86_0
lcms2                     2.11                 h396b838_0
ld_impl_linux-64          2.33.1               h53a641e_7
leather                   0.3.3                    py38_0
libarchive                3.4.2                h62408e4_0
libedit                   3.1.20191231         h14c3975_1
libffi                    3.2.1                hd88cf55_4
libgcc-ng                 9.1.0                hdf63c60_0
libgfortran-ng            7.3.0                hdf63c60_0
libiconv                  1.15                 h63c8f33_5
liblief                   0.10.1               he6710b0_0
libllvm9                  9.0.1                h4a3c616_1
libpng                    1.6.37               hbc83047_0
libprotobuf               3.12.3               hd408876_0
libsodium                 1.0.18               h7b6447c_0
libstdcxx-ng              9.1.0                hdf63c60_0
libtiff                   4.1.0                h2733197_1
libuuid                   1.0.3                h1bed415_2
libxcb                    1.14                 h7b6447c_0
libxml2                   2.9.10               he19cac6_1
libxslt                   1.1.34               hc22bd24_0
lightgbm                  2.3.0            py38he6710b0_0
line_profiler             2.1.2            py38h7b6447c_0
llvmlite                  0.33.0           py38hc6ec683_1
locket                    0.2.0                    py38_1
lxml                      4.5.2            py38hefd8a0e_0
lz4-c                     1.9.2                he6710b0_0
magma-cuda110             2.5.2                         1    pytorch
mako                      1.1.3                      py_0
markdown                  3.1.1                    py38_0
markupsafe                1.1.1            py38h7b6447c_0
matplotlib                3.2.2                         0
matplotlib-base           3.2.2            py38hef1b27d_0
mccabe                    0.6.1                    py38_1
metis                     5.1.0                hf484d3e_4
mistune                   0.8.4           py38h7b6447c_1000
mkl                       2020.1                      217
mkl-include               2020.1                      217
mkl-service               2.3.0            py38he904b0f_0
mkl_fft                   1.1.0            py38h23d657b_0
mkl_random                1.1.1            py38h0573a6f_0
mock                      4.0.2                      py_0
more-itertools            8.4.0                      py_0
mosek                     9.2.16                   py38_0    mosek
mpc                       1.1.0                h10f8cd9_1
mpfr                      4.0.2                hb69a4c5_1
mpmath                    1.1.0                    py38_0
msgpack-python            1.0.0            py38hfd86e86_1
multi_key_dict            2.0.3                      py_0    conda-forge
murmurhash                1.0.2            py38he6710b0_0
mysql-connector-c         6.1.11               h597af5e_1
mysqlclient               2.0.1            py38he6710b0_0
nbconvert                 5.6.1                    py38_0
nbformat                  5.0.7                      py_0
ncurses                   6.2                  he6710b0_1
networkx                  2.4                        py_1
ninja                     1.9.0            py38hfd86e86_0
notebook                  6.0.3                    py38_0
numba                     0.50.1           py38h0573a6f_1
numexpr                   2.7.1            py38h423224d_0
numpy                     1.18.5           py38ha1c710e_0
numpy-base                1.18.5           py38hde5b4d6_0
oauthlib                  3.1.0                      py_0
objgraph                  3.4.1              pyh9f0ad1d_0    conda-forge
olefile                   0.46                       py_0
openpyxl                  3.0.4                      py_0
openssl                   1.1.1g               h7b6447c_0
opt_einsum                3.1.0                      py_0
packaging                 20.4                       py_0
pandas                    1.0.5            py38h0573a6f_0
pandoc                    2.10                          0
pandocfilters             1.4.2                    py38_1
pango                     1.42.4               h049681c_0
paramiko                  2.7.1                      py_0
parse                     1.14.0                     py_0    conda-forge
parsedatetime             2.4                      py38_0
parso                     0.7.0                      py_0
partd                     1.1.0                      py_0
patchelf                  0.11                 he6710b0_0
patsy                     0.5.1                    py38_0
pcre                      8.44                 he6710b0_0
pexpect                   4.8.0                    py38_0
pickleshare               0.7.5                 py38_1000
pillow                    7.2.0            py38hb39fc2d_0
pip                       20.1.1                   py38_1
pixman                    0.40.0               h7b6447c_0
pkginfo                   1.5.0.1                  py38_0
plac                      0.9.6                    py38_1
plotly                    4.8.2                      py_0
pluggy                    0.13.1                   py38_0
preshed                   3.0.2            py38he6710b0_1
prometheus_client         0.8.0                      py_0
prompt-toolkit            3.0.5                      py_0
prompt_toolkit            3.0.5                         0
protobuf                  3.12.3           py38he6710b0_0
psutil                    5.7.0            py38h7b6447c_0
ptyprocess                0.6.0                    py38_0
py                        1.9.0                      py_0
py-lief                   0.10.1           py38h403a769_0
pyasn1                    0.4.8                      py_0
pyasn1-modules            0.2.7                      py_0
pybind11                  2.5.0            py38hfd86e86_0
pycodestyle               2.6.0                      py_0
pycosat                   0.6.3            py38h7b6447c_1
pycparser                 2.20                       py_2
pyflakes                  2.2.0                      py_0
pygments                  2.6.1                      py_0
pyhamcrest                2.0.2                      py_0
pyjwt                     1.7.1                    py38_0
pynacl                    1.4.0            py38h7b6447c_1
pyopenssl                 19.1.0                     py_1
pyparsing                 2.4.7                      py_0
pyqt                      5.9.2            py38h05f1152_4
pyrsistent                0.16.0           py38h7b6447c_0
pysocks                   1.7.1                    py38_0
pytest                    5.4.3                    py38_0
python                    3.8.2                hcf32534_0
python-dateutil           2.8.1                      py_0
python-editor             1.0.4                      py_0
python-jenkins            1.7.0                      py_0    conda-forge
python-libarchive-c       2.9                        py_0
python-slugify            3.0.4                      py_0
python_abi                3.8                      1_cp38    conda-forge
pytimeparse               1.1.8                    py38_0
pytz                      2020.1                     py_0
pywavelets                1.1.1            py38h7b6447c_0
pyyaml                    5.3.1            py38h7b6447c_1
pyzmq                     19.0.1           py38he6710b0_1
qt                        5.9.7                h5867ecd_1
qtconsole                 4.7.5                      py_0
qtpy                      1.9.0                      py_0
readline                  8.0                  h7b6447c_0
redis-py                  3.5.3                      py_0
requests                  2.24.0                     py_0
requests-oauthlib         1.3.0                      py_0
retrying                  1.3.3                      py_2
ripgrep                   11.0.2               he32d670_0
rq                        1.2.2                      py_0    conda-forge
rsa                       4.0                        py_0
ruamel_yaml               0.15.87          py38h7b6447c_1
scikit-image              0.16.2           py38h0573a6f_0
scikit-learn              0.23.1           py38h423224d_0
scipy                     1.5.0            py38h0b6359f_0
seaborn                   0.10.1                     py_0
send2trash                1.5.0                    py38_0
service_identity          18.1.0                     py_0
setuptools                49.2.0                   py38_0
simplejson                3.17.0           py38h7b6447c_0
sip                       4.19.13          py38he6710b0_0
six                       1.15.0                     py_0
snakeviz                  2.0.1                    py38_0
sortedcontainers          2.2.2                      py_0
soupsieve                 2.0.1                      py_0
spacy                     2.3.1            py38hfd86e86_0
sqlalchemy                1.3.18           py38h7b6447c_0
sqlite                    3.32.3               h62c20be_0
sqlparse                  0.3.1                      py_0
srsly                     1.0.2            py38he6710b0_0
statsmodels               0.11.1           py38h7b6447c_0
suitesparse               5.2.0                h9e4a6bb_0
sympy                     1.6.1                    py38_0
tabulate                  0.8.3                    py38_0
tbb                       2020.0               hfd86e86_0
tblib                     1.6.0                      py_0
tensorboard               2.2.1              pyh532a8cf_0
tensorboard-plugin-wit    1.6.0                      py_0
tensorboardx              2.1                        py_0    conda-forge
tensorflow                2.2.0           mkl_py38h6d3daf0_0
tensorflow-base           2.2.0           mkl_py38h5059a2d_0
tensorflow-estimator      2.2.0              pyh208ff02_0
termcolor                 1.1.0                    py38_1
terminado                 0.8.3                    py38_0
testpath                  0.4.4                      py_0
text-unidecode            1.3                        py_0
thinc                     7.4.1            py38hfd86e86_0
threadpoolctl             2.1.0              pyh5ca1d4c_0
tk                        8.6.10               hbc83047_0
toolz                     0.10.0                     py_0
tornado                   6.0.4            py38h7b6447c_1
tqdm                      4.47.0                     py_0
traitlets                 4.3.3                    py38_0
twisted                   20.3.0           py38h7b6447c_0
txaio                     20.3.1                     py_0    conda-forge
typing_extensions         3.7.4.2                    py_0
tzlocal                   2.0.0                    py38_0
ujson                     3.0.0            py38he6710b0_0
unidecode                 1.1.1                      py_0
urllib3                   1.25.9                     py_0
voluptuous                0.11.7                     py_0    conda-forge
wasabi                    0.6.0                      py_0
wcwidth                   0.2.5                      py_0
webencodings              0.5.1                    py38_1
websocket-client          0.57.0                   py38_1
werkzeug                  1.0.1                      py_0
wheel                     0.34.2                   py38_0
widgetsnbextension        3.5.1                    py38_0
wrapt                     1.12.1           py38h7b6447c_1
xarray                    0.16.0                     py_0
xlrd                      1.2.0                      py_0
xz                        5.2.5                h7b6447c_0
yaml                      0.2.5                h7b6447c_0
zeromq                    4.3.2                he6710b0_2
zict                      2.0.0                      py_0
zipp                      3.1.0                      py_0
zlib                      1.2.11               h7b6447c_3
zope                      1.0                      py38_1
zope.event                4.4                      py38_0
zope.interface            4.7.1            py38h7b6447c_0
zstd                      1.4.5                h0b5b093_0
```
</details>



<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info
     active environment : /dev/shm/envs/hello                                                                                               
    active env location : /dev/shm/envs/hello                                                                                               
            shell level : 3                                                                                                                   
       user config file : /home/user/.condarc                                                                                             
 populated config files : /path/somewhere/miniconda3/.condarc                                                                               
                          /home/user/.condarc                                                                                             
          conda version : 4.8.3                                                                                                               
    conda-build version : not installed                                                                                                       
         python version : 3.7.6.final.0                                                                                                       
       virtual packages : __glibc=2.17                                                                                                        
       base environment : /path/somewhere/miniconda3  (read only)                                                                           
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64                                                                        
                          https://repo.anaconda.com/pkgs/main/noarch                                                                          
                          https://repo.anaconda.com/pkgs/r/linux-64                                                                           
                          https://repo.anaconda.com/pkgs/r/noarch                                                                             
                          https://conda.anaconda.org/conda-forge/linux-64                                                                     
                          https://conda.anaconda.org/conda-forge/noarch                                                                       
                          https://conda.anaconda.org/prometeia/linux-64                                                                       
                          https://conda.anaconda.org/prometeia/noarch                                                                         
                          https://conda.anaconda.org/mosek/linux-64                                                                           
                          https://conda.anaconda.org/mosek/noarch                                                                             
          package cache : /path/somewhere/miniconda3/pkgs                                                                                   
                          /home/user/.conda/pkgs                                                                                          
       envs directories : /path/somewhere/miniconda3/envs                                                                                   
                          /home/user/.conda/envs                                                                                          
               platform : linux-64                                                                                                            
             user-agent : conda/4.8.3 requests/2.23.0 CPython/X Linux/Y-stuff/Z glibc/W                
                UID:GID : 41652:40035                                                                                                         
             netrc file : None                                                                                                                
           offline mode : False                                                                                                               
```
</details>

**Comment by isuruf** [comment]
Why did you delete the issue template and not answer the information required there?

**Comment by chebee7i** [comment]
If it's critical, I can re-add. But there is sometimes peronsal-ish information in those that I'd don't want to spend time editing.

**Comment by isuruf** [comment]
It is critical. That's why we are asking for that information. Please spend time editing before asking us to spend time helping you out.

**Comment by chebee7i** [comment]
Updated. Is the issue that conda_build must be installed in the base environment?

**Comment by isuruf** [comment]
You need `conda-forge` with higher priority than defaults. You have it lower.

**Comment by chebee7i** [comment]
Thanks. And for my own education, did you derive this from the package list and seeing non-conda-forge version of conda-build?  To test this, I just created a new environment, but even after activating into it, there is still no `CONDA_BUILD_SYSROOT`.

```console
$ conda activate /dev/shm/envs/hello2
$ echo $CONDA_PREFIX
/dev/shm/envs/hello2
$ echo "x${CONDA_BUILD_SYSROOT}x"
xx
```

<br/>
Install 
<details>

```
$ conda create -p /dev/shm/envs/hello2 conda-build python=3.8 -c conda-forge
Collecting package metadata (current_repodata.json): done
Solving environment: done


## Package Plan ##

  environment location: /dev/shm/envs/hello2

  added / updated specs:
    - conda-build
    - python=3.8


The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    _libgcc_mutex-0.1          |      conda_forge           3 KB  conda-forge
    _openmp_mutex-4.5          |            1_gnu          22 KB  conda-forge
    beautifulsoup4-4.9.1       |   py38h32f6830_0         163 KB  conda-forge
    brotlipy-0.7.0             |py38h1e0a361_1000         346 KB  conda-forge
    bzip2-1.0.8                |       h516909a_2         396 KB  conda-forge
    ca-certificates-2020.6.20  |       hecda079_0         145 KB  conda-forge
    certifi-2020.6.20          |   py38h32f6830_0         151 KB  conda-forge
    cffi-1.14.1                |   py38h5bae8af_0         225 KB  conda-forge
    chardet-3.0.4              |py38h32f6830_1006         170 KB  conda-forge
    conda-4.8.4                |   py38h32f6830_2         3.0 MB  conda-forge
    conda-build-3.19.2         |   py38h32f6830_3         528 KB  conda-forge
    conda-package-handling-1.7.0|   py38h1e0a361_4         2.0 MB  conda-forge
    cryptography-3.0           |   py38h766eaa4_0         632 KB  conda-forge
    filelock-3.0.12            |     pyh9f0ad1d_0          10 KB  conda-forge
    glob2-0.7                  |             py_0          11 KB  conda-forge
    icu-67.1                   |       he1b5a44_0        12.9 MB  conda-forge
    idna-2.10                  |     pyh9f0ad1d_0          52 KB  conda-forge
    jinja2-2.11.2              |     pyh9f0ad1d_0          93 KB  conda-forge
    ld_impl_linux-64-2.34      |       hc38a660_9         612 KB  conda-forge
    libarchive-3.3.3           |    h3a8160c_1008         1.5 MB  conda-forge
    libffi-3.2.1               |    he1b5a44_1007          47 KB  conda-forge
    libgcc-ng-9.3.0            |      h24d8f2e_14         7.8 MB  conda-forge
    libgomp-9.3.0              |      h24d8f2e_14         374 KB  conda-forge
    libiconv-1.15              |    h516909a_1006         2.0 MB  conda-forge
    liblief-0.10.1             |       he1b5a44_0         2.0 MB  conda-forge
    libstdcxx-ng-9.3.0         |      hdf63c60_14         4.0 MB  conda-forge
    libxml2-2.9.10             |       h72b56ed_2         1.3 MB  conda-forge
    lz4-c-1.9.2                |       he1b5a44_1         226 KB  conda-forge
    lzo-2.10                   |    h14c3975_1000         319 KB  conda-forge
    markupsafe-1.1.1           |   py38h1e0a361_1          26 KB  conda-forge
    ncurses-6.2                |       he1b5a44_1         993 KB  conda-forge
    openssl-1.1.1g             |       h516909a_1         2.1 MB  conda-forge
    patchelf-0.11              |       he1b5a44_0          78 KB  conda-forge
    pip-20.2.2                 |             py_0         1.1 MB  conda-forge
    pkginfo-1.5.0.1            |             py_0          21 KB  conda-forge
    psutil-5.7.2               |   py38h1e0a361_0         337 KB  conda-forge
    py-lief-0.10.1             |   py38h950e882_0         1.3 MB  conda-forge
    pycosat-0.6.3              |py38h1e0a361_1004         107 KB  conda-forge
    pycparser-2.20             |     pyh9f0ad1d_2          94 KB  conda-forge
    pyopenssl-19.1.0           |             py_1          47 KB  conda-forge
    pysocks-1.7.1              |   py38h32f6830_1          27 KB  conda-forge
    python-3.8.5               |h4d41432_2_cpython        71.0 MB  conda-forge
    python-libarchive-c-2.9    |           py38_0          54 KB  conda-forge
    pytz-2020.1                |     pyh9f0ad1d_0         227 KB  conda-forge
    pyyaml-5.3.1               |   py38h1e0a361_0         190 KB  conda-forge
    readline-8.0               |       he28a2e2_2         281 KB  conda-forge
    requests-2.24.0            |     pyh9f0ad1d_0          47 KB  conda-forge
    ripgrep-12.1.1             |       h516909a_0         1.8 MB  conda-forge
    ruamel_yaml-0.15.80        |py38h1e0a361_1001         270 KB  conda-forge
    setuptools-49.4.0          |   py38h32f6830_0         952 KB  conda-forge
    six-1.15.0                 |     pyh9f0ad1d_0          14 KB  conda-forge
    soupsieve-2.0.1            |   py38h32f6830_0          57 KB  conda-forge
    sqlite-3.32.3              |       hcee41ef_1         1.4 MB  conda-forge
    tk-8.6.10                  |       hed695b0_0         3.2 MB  conda-forge
    tqdm-4.48.2                |     pyh9f0ad1d_0          53 KB  conda-forge
    urllib3-1.25.10            |             py_0          92 KB  conda-forge
    xz-5.2.5                   |       h516909a_1         343 KB  conda-forge
    yaml-0.2.5                 |       h516909a_0          82 KB  conda-forge
    zlib-1.2.11                |    h516909a_1007         106 KB  conda-forge
    zstd-1.4.5                 |       h6597ccf_2         712 KB  conda-forge
    ------------------------------------------------------------
                                           Total:       127.8 MB

<snip>
```

</details>

<br/>
Environment (conda list): 
<details>

```
$ conda info
# packages in environment at /dev/shm/envs/hello2:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       1_gnu    conda-forge
beautifulsoup4            4.9.1            py38h32f6830_0    conda-forge
brotlipy                  0.7.0           py38h1e0a361_1000    conda-forge
bzip2                     1.0.8                h516909a_2    conda-forge
ca-certificates           2020.6.20            hecda079_0    conda-forge
certifi                   2020.6.20        py38h32f6830_0    conda-forge
cffi                      1.14.1           py38h5bae8af_0    conda-forge
chardet                   3.0.4           py38h32f6830_1006    conda-forge
conda                     4.8.4            py38h32f6830_2    conda-forge
conda-build               3.19.2           py38h32f6830_3    conda-forge
conda-package-handling    1.7.0            py38h1e0a361_4    conda-forge
cryptography              3.0              py38h766eaa4_0    conda-forge
filelock                  3.0.12             pyh9f0ad1d_0    conda-forge
glob2                     0.7                        py_0    conda-forge
icu                       67.1                 he1b5a44_0    conda-forge
idna                      2.10               pyh9f0ad1d_0    conda-forge
jinja2                    2.11.2             pyh9f0ad1d_0    conda-forge
ld_impl_linux-64          2.34                 hc38a660_9    conda-forge
libarchive                3.3.3             h3a8160c_1008    conda-forge
libffi                    3.2.1             he1b5a44_1007    conda-forge
libgcc-ng                 9.3.0               h24d8f2e_14    conda-forge
libgomp                   9.3.0               h24d8f2e_14    conda-forge
libiconv                  1.15              h516909a_1006    conda-forge
liblief                   0.10.1               he1b5a44_0    conda-forge
libstdcxx-ng              9.3.0               hdf63c60_14    conda-forge
libxml2                   2.9.10               h72b56ed_2    conda-forge
lz4-c                     1.9.2                he1b5a44_1    conda-forge
lzo                       2.10              h14c3975_1000    conda-forge
markupsafe                1.1.1            py38h1e0a361_1    conda-forge
ncurses                   6.2                  he1b5a44_1    conda-forge
openssl                   1.1.1g               h516909a_1    conda-forge
patchelf                  0.11                 he1b5a44_0    conda-forge
pip                       20.2.2                     py_0    conda-forge
pkginfo                   1.5.0.1                    py_0    conda-forge
psutil                    5.7.2            py38h1e0a361_0    conda-forge
py-lief                   0.10.1           py38h950e882_0    conda-forge
pycosat                   0.6.3           py38h1e0a361_1004    conda-forge
pycparser                 2.20               pyh9f0ad1d_2    conda-forge
pyopenssl                 19.1.0                     py_1    conda-forge
pysocks                   1.7.1            py38h32f6830_1    conda-forge
python                    3.8.5           h4d41432_2_cpython    conda-forge
python-libarchive-c       2.9                      py38_0    conda-forge
python_abi                3.8                      1_cp38    conda-forge
pytz                      2020.1             pyh9f0ad1d_0    conda-forge
pyyaml                    5.3.1            py38h1e0a361_0    conda-forge
readline                  8.0                  he28a2e2_2    conda-forge
requests                  2.24.0             pyh9f0ad1d_0    conda-forge
ripgrep                   12.1.1               h516909a_0    conda-forge
ruamel_yaml               0.15.80         py38h1e0a361_1001    conda-forge
setuptools                49.4.0           py38h32f6830_0    conda-forge
six                       1.15.0             pyh9f0ad1d_0    conda-forge
soupsieve                 2.0.1            py38h32f6830_0    conda-forge
sqlite                    3.32.3               hcee41ef_1    conda-forge
tk                        8.6.10               hed695b0_0    conda-forge
tqdm                      4.48.2             pyh9f0ad1d_0    conda-forge
urllib3                   1.25.10                    py_0    conda-forge
wheel                     0.34.2                     py_1    conda-forge
xz                        5.2.5                h516909a_1    conda-forge
yaml                      0.2.5                h516909a_0    conda-forge
zlib                      1.2.11            h516909a_1007    conda-forge
zstd                      1.4.5                h6597ccf_2    conda-forge
```
</details>

**Comment by isuruf** [comment]
> And for my own education, did you derive this from the package list and seeing non-conda-forge version of conda-build? 

From ` channel URLs :` output in `conda info`

As to `CONDA_BUILD_SYSROOT`, you need to install the compilers as well. `conda install gcc_linux-64 gxx_linux-64`

**Comment by chebee7i** [comment]
I see. We actually provide compilers externally from conda, due to other build systems that are also in use. Not sure the immediate impact, but that's a downstream concern. Thanks again!

**Comment by kkraus14** [comment]
> I see. We actually provide compilers externally from conda, due to other build systems that are also in use. Not sure the immediate impact, but that's a downstream concern. Thanks again!

Sounds like we should enable this use case for the conda package still where if `CONDA_BUILD_SYSROOT` isn't set we can just skip trying to symlink `libcuda.so` into a sysroot directory.

**Comment by chebee7i** [comment]
Yeah that would be nice!  

Also, just to circle back around to the ultimate root cause here: it was not necessary that `conda-forge` have higher priority than `defaults`.  I was able to create an environment which included `conda-build`, `gcc_linux_64` and every other dependency all taken from `defaults`. In that environment, `CONDA_BUILD_SYSROOT` was defined (due to inclusion of `gcc_linux_64`).  After that, installing `nvcc_linux_64` proceeded without error.

**Comment by tvandera** [comment]
I also fell into this bug. Would be nice if it could be fixed. Tnx T.

**Comment by izahn** [comment]
I just spent several hours stuck on this, and definitely consider it a bug. If nvcc needs gcc in order to function then it should declare it as a dependency.

**Comment by kkraus14** [comment]
Should be fixed by #84
